app/kubernetes/__module.js

```diff
@@ -6,19 +6,25 @@ angular.module('portainer.kubernetes', ['portainer.app']).config([
     const kubernetes = {
       name: 'kubernetes',
       url: '/kubernetes',
-      parent: 'root',
+      parent: 'endpoint',
       abstract: true,
       resolve: {
-        endpointID: [
-          'EndpointProvider',
-          '$state',
-          function (EndpointProvider, $state) {
-            const id = EndpointProvider.endpointID();
-            if (!id) {
-              return $state.go('portainer.home');
+        /* @ngInject */
+        endpointCheck($async, endpoint, EndpointProvider, KubernetesHealthService, StateManager) {
+          return $async(async () => {
+            if (endpoint.Type === 7) {
+              try {
```

Comment:
in `develop` this is done only for edge, but maybe it should happen for all endpoints? even on the root endpoint resolver?